cdap-docs/developer-manual/source/metadata/programmatic-metadata.rst

```diff
@@ -0,0 +1,85 @@
+.. meta::
+    :author: Cask Data, Inc.
+    :copyright: Copyright © 2018 Cask Data, Inc.
+
+.. _programmatic-metadata:
+
+==================================
+Metadata from Programs and Plugins
+==================================
+
+.. highlight:: java
+
+Metadata can be accessed from *Programs* and *Plugins*. Currently, reading metadata is not supported while CDAP is running in *cloud mode* but metadata can be added, updated or deleted in all modes.
+
+.. _metadata-programs:
+
+Programs
+========
+
+Metadata can be accessed from *MapReduce*, *Spark*, *Workers* and *Service* through their individual program context.
+Metadata operations are not supported at individual tasks and should be performed only during the
```

Comment:
is this more a best practice? Or does the API not allow this to happen?

Reply:
We don't even expose these api to task level so I will just remove this line.